### PR TITLE
Plot lib change for summer

### PR DESCRIPTION
**Pull request by sorawee**
@SDooman @jpolitz 

1. convert pie-chart to list style in plot-list
2. add checks preventing unequal length list for those that require the check
3. fix the bug when trivial data is provided (0 or 1 data point)
4. add freq-bar-chart

### CODE REVIEW

File: src/web/arr/trove/plot-list.arr [hunk 1]
```diff
@@ -197,83 +201,106 @@ where:
...
+    [list: 1, 2, 3],
+    [list: 1, 2, 3],
+    default-options) does-not-raise
 end
 
 fun bar-chart(
-    labels :: List<String>,
-    values :: List<Number>,
-    options-generator :: WrappedBarChartWindowOptions) -> IM.Image block:
+    label :: List<String>,
```
Comment (sorawee) [inline src/web/arr/trove/plot-list.arr]: @SDooman here

File: src/web/arr/trove/plot-list.arr [hunk 2]
```diff
@@ -197,83 +201,106 @@ where:
...
-fun pie-chart(tab :: Table, options-generator :: WrappedPieChartWindowOptions) -> IM.Image block:
-  doc: 'Consume a table with two columns: `label` and `value`, and show a pie-chart'
-  when not(tab._header-raw-array =~ [raw-array: 'label', 'value']):
-    raise('pie-chart: expect a table with columns named `label` and `value`')
-  end
-  when raw-array-length(tab._rows-raw-array) == 0:
-    raise('pie-chart: expect the table to have at least one row')
-  end
+fun pie-chart(
+    label :: List<String>,
```
Comment (sorawee) [inline src/web/arr/trove/plot-list.arr]: and here @SDooman

**Comment by SDooman** [comment]
looks great to me!

**Comment by schanzer** [comment]
Whoever merges this: make sure to also close #985 when you do.

**Comment by jpolitz** [comment]
If I'm reading this right, it changes function signatures.  We really ruined the day of a few physics teachers the last time we did that with plotting.  They're probably in summer now, but this will break their code from last year.

Is it intentional that this interface gets updated, and not feasible to introduce new functions with updated signatures?

**Comment by jpolitz** [comment]
Also, is there a corresponding update on the docs repo?  I checked a few branches but don't see it.

**Comment by sorawee** [comment]
@jpolitz the function signature change goes to the `plot-list.arr` file which is a new file and is right now not documented, so no one has used it, and it won't break anyone's code. The `plot.arr` is the one that physics teacher would have used, and that is left unchanged (in fact, that's precisely the reason why we create `plot-list.arr`).

Yes, I will need to do the documentation.

**Comment by jpolitz** [comment]
Thanks for the explanation @sorawee – apologies for being overly cautious.

**Pull request merged**